### DISCUSSION TITLE
wsa virtwifi saved network changes dns1 entry to vEthernet (wsl) gateway address

### DISCUSSION THREAD

**Discussion by myusrn** [LSPosed/MagiskOnWSALocal#199] [Q&A] [answered]
Question
---
Anyone else seeing this and have insights on how you prevent it from happening permanently in the wsa configuration settings?

Background
---
I'm finding an issue with my current wsa with magisk and mindthegapps install, which i suspect might affect standard issue wsa installs as well, arising after some period of use and system sleep/restore events.  

Figured i'd ask here since this group understands wsa better than what is available in other forums. 

What happens is apps are still running fine but any network active results in a failure. A search hit led me to find that this arises from the wsa virtwifi saved network settings end up with its dns1 entry pointing at the vEthernet (wsl) gateway address.  See `Get-NetIPAddress` or `ipconfig /all` for vEthernet (wsl) details which usually has a 172.16.0.0/12 or 192.168.0.0/16 private network address.

The fix is to run `& \adb.exe connect 127.0.0.1:58526` followed by `& .\adb.exe shell am start -n 'com.android.settings/.Settings\$WifiSettingsActivity'` to launch the wsa network settings UI and change the dns1 entry to something like google's 8.8.8.8 dns server given the dns2 entry is always showing me the corresponding 8.8.4.4 alternate dns server address.

**Comment by Howard20181** [comment, marked as answer]
If you don't enable the advanced network it will use NAT, WSA through vEthernet networking, WSA's virtwifi's DNS is Windows NAT, normally it can resolve the domain name

**Comment by myusrn** [comment]
@Howard20181 thank you for the response.  

I'm hearing that if wsa settings | advanced networking is set to the default of `off` as mine is then it is expected that the VirtWifi interface's dns1 setting will be set to point to the `vEthernet (WSL)` interface that acts as the NAT for wsa as well wsl distros. 

Just prior to wsa networking dns1 no longer successfully resolving addresses i get a pop-up dialog, which i'll grab the details of next time it arises and paste here, saying something to the tune of wsa networking had to be shutdown click here to restart subsystem.  It's just after i click that restart subsystem button that networking in apps stops working and only way i found to recover it, without doing a reboot, is by pointing dns1 setting at an external dns server vs the wsa/wsl nat interface.

Does that sound like anything you've seen before?  Is the resolution here that i enable advanced networking even if i don't need ability it advertises as providing, which is be able to see and communicate with other devices on my local private network?

**Reply by Howard20181** [reply]
> dns1 no longer successfully resolving addresses

I have not encountered this problem for the time being, maybe it is a new bug?

> something to the tune of wsa networking had to be shutdown click here to restart subsystem.

If advanced networking is enabled and WSA is started, this prompt will pop up if tun is enabled on Windows like [Wintun](https://www.wintun.net/) or something like that.

**Reply by myusrn** [reply]
This morning i was able to capture the dialog that i found previously was, perhaps coincidentally, displayed around the same time my wsa networking would stop working.  Apparently due to dns1 name resolution issues given it could be restored by going into wsa network settings and changing VirtWifi dns1 to a public dns server like google's 8.8.8.8 vs the vEnternet (wsl) nat / gateway interface ip address default.

![image](https://user-images.githubusercontent.com/3043446/196999185-137b58c0-c217-4add-afe8-008b6d95eede.png)
where "Learn more" points to https://go.microsoft.com/fwlink/?linkid=2186250 -> [Manage settings for mobile apps on Windows (microsoft.com)](https://support.microsoft.com/en-us/windows/manage-settings-for-mobile-apps-on-windows-000f97e8-8c20-490e-9ef4-cd90d903f847).

Also perhaps of interest is I updated from 22h2 dev insider 25227.1000 last night and now when i visit wsa networking [ using  `& \adb.exe connect 127.0.0.1:58526` followed by `& .\adb.exe shell am start -n 'com.android.settings/.Settings\$WifiSettingsActivity'` ] it's no longer showing a VirtWifi interface and instead showing the same wifi connection ssid and details that my windows host is using.  It's also showing me all the other locally available wifi networks.  It would appear the new win11 dev insider and/or wsa preview install components have changed have network connectivity is provided.  Overall, the performance of this new insider install experience subjectively feels snappier.

**Reply by aogg** [reply]
+1

**Comment by l-l-l-l-l-l** [comment]
I have updated to the latest build of WSA, but the option to enable advanced network settings doesn't appear in my settings. Is there a way to enable advanced network settings through adb?

**Reply by Howard20181** [reply]
No, please upgrade Windows 11 to 22H2

**Reply by l-l-l-l-l-l** [reply]
I have just updated and activated advanced networking. The moment I turn WSA off and on again, it still forgets the settings.

**Reply by l-l-l-l-l-l** [reply]
OK, now it suddently it forgot the existence of vitwifi and reads my wifi list as is. Fixed, i suppose.

**Comment by myusrn** [comment]
What i found was if wsa advanced networking is enabled, you see actual wifi ssid values and connections.  The wifi ssid that your windows host is connected to will be enabled by default. 

If wsa advanced networking is disabled, or gets automatically disabled when an unsupported vpn client connection is detected, then you go back to having the VirtWifi interface which uses the "vEthernet (wsl)" interface as its gateway and dns.
